### PR TITLE
add String datetime stats collector

### LINKED ISSUE #7786: StringColumnPreIndexStatsCollector doesnt work for datetime
datetime fields are not necessarily sorted by string ordering. 

adding StringDatetimeColumnPreIndexStatsCollector to handle the ordering issue

### PR DESCRIPTION
**Pull request by walterddr**
This fixes #7786

### CODE REVIEW

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/stats/StringDateTimeColumnPreIndexStatsCollector.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  public StringDateTimeColumnPreIndexStatsCollector(String column, DateTimeFieldSpec fieldSpec,
+      StatsCollectorConfig statsCollectorConfig) {
+    super(column, statsCollectorConfig);
+    _dateTimeFormatter = new DateTimeFormatSpec(fieldSpec.getFormat()).getDateTimeFormatter();
+  }
+
+  @Override
+  public void seal() {
+    _sortedValues = _values.toArray(new String[0]);
+    Arrays.sort(_sortedValues, new StringDateTimeComparator(_dateTimeFormatter));
```
Comment (richardstartin) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/stats/StringDateTimeColumnPreIndexStatsCollector.java:41]: I'm curious how long this takes for 10M+ dates, would it be better to parse and store in `collect` instead?

Comment (walterddr) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/stats/StringDateTimeColumnPreIndexStatsCollector.java:41]: good question. i dunno but all other stats collector is doing the same. i can follow up with another PR. this one is more of a feature/bugfix.

Comment (richardstartin) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/stats/StringDateTimeColumnPreIndexStatsCollector.java:41]: Oh, I mean the others don't have an expensive comparator, but this one has, and the only reason for that is to reuse code for handling strings.

**Comment by walterddr** [comment]
closing. this is not a valid working solution

**Pull request closed without merging**